From 697a7ba791fce2ffab2ed723d909a89684019e3d Mon Sep 17 00:00:00 2001
From: "Tuan M. Hoang" <tmhoang@flatglobe.org>
Date: Sun, 19 Feb 2017 17:03:33 +0000
Subject: [PATCH] main/gcc : when building gcc-java on s390x, libgcj.so
 complains about undefined reference to __data_start. Using
 SEARCH_FOR_DATA_START instead solved it. Credit to Andrew Haley at RedHat.

---
 boehm-gc/include/private/gcconfig.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/boehm-gc/include/private/gcconfig.h b/boehm-gc/include/private/gcconfig.h
index 4ca6a80..a15fff6 100644
--- a/boehm-gc/include/private/gcconfig.h
+++ b/boehm-gc/include/private/gcconfig.h
@@ -1837,8 +1837,7 @@
 #       define OS_TYPE "LINUX"
 #       define LINUX_STACKBOTTOM
 #       define DYNAMIC_LOADING
-	extern int __data_start[];
-#       define DATASTART ((ptr_t)(__data_start))
+#       define SEARCH_FOR_DATA_START
     extern int _end[];
 #   define DATAEND (_end)
 #   define CACHE_LINE_SIZE 256
-- 
2.11.1

